### ISSUE TITLE
Trouble getting aliasing to work

### ISSUE THREAD

**Issue by jerrywgray** [reactiverse/es4x#103]
Like the original issue that spawned the module aliases ticket I'm working on getting a graphql service working in javascript but also available as a verticle to the rest of our backend. I'm using Apollo Server and wrote most of an adapter like their express adapter but using es4x instead.

However there's one file(so far) that's imported directly by apollo-server-core(node-fetch) that was never compiled down to ES5 code. It still contains a class so nashorn is throwing a parse error.

I tried just compiling the file to get it to standard ES5 and then used the alias directive to point at that file but when es4x builds the project it's still importing the file from node_modules instead of my build directory. I looked at the code in es4x where the alias gets handled and can't see why it wouldn't get picked up. I have no idea how to set up a debugger to step into that import code though since es4x is entirely on the java classpath so I don't even know where I could even stick a breakpoint.

Have you noticed any issues with the alias? Or maybe I'm just defining it incorrectly? I don't see any parts of the stack trace that indicate it tries to load the aliased file at all.

Stack trace:
```
java.lang.RuntimeException: node_modules/node-fetch/lib/index.js:21:0 ES6 class declarations and expressions are not yet implemented
class Blob {
^
        at jdk.scripting.nashorn/jdk.nashorn.internal.codegen.Lower.throwNotImplementedYet(Lower.java:875)
        at jdk.scripting.nashorn/jdk.nashorn.internal.codegen.Lower.enterClassNode(Lower.java:706)
        at jdk.scripting.nashorn/jdk.nashorn.internal.ir.ClassNode.accept(ClassNode.java:131)
        at jdk.scripting.nashorn/jdk.nashorn.internal.ir.VarNode.accept(VarNode.java:197)
        at jdk.scripting.nashorn/jdk.nashorn.internal.ir.Node.accept(Node.java:286)
        at jdk.scripting.nashorn/jdk.nashorn.internal.ir.Block.accept(Block.java:235)
        at jdk.scripting.nashorn/jdk.nashorn.internal.ir.LexicalContextNode$Acceptor.accept(LexicalContextNode.java:57)
        at jdk.scripting.nashorn/jdk.nashorn.internal.ir.Block.accept(Block.java:511)
        at jdk.scripting.nashorn/jdk.nashorn.internal.ir.FunctionNode.accept(FunctionNode.java:405)
        at jdk.scripting.nashorn/jdk.nashorn.internal.ir.LexicalContextNode$Acceptor.accept(LexicalContextNode.java:57)
        at jdk.scripting.nashorn/jdk.nashorn.internal.ir.LexicalContextExpression.accept(LexicalContextExpression.java:47)
        at jdk.scripting.nashorn/jdk.nashorn.internal.ir.FunctionNode.accept(FunctionNode.java:57)
        at jdk.scripting.nashorn/jdk.nashorn.internal.ir.ExpressionStatement.accept(ExpressionStatement.java:82)
        at jdk.scripting.nashorn/jdk.nashorn.internal.ir.Node.accept(Node.java:286)
        at jdk.scripting.nashorn/jdk.nashorn.internal.ir.Block.accept(Block.java:235)
        at jdk.scripting.nashorn/jdk.nashorn.internal.ir.LexicalContextNode$Acceptor.accept(LexicalContextNode.java:57)
        at jdk.scripting.nashorn/jdk.nashorn.internal.ir.Block.accept(Block.java:511)
        at jdk.scripting.nashorn/jdk.nashorn.internal.ir.FunctionNode.accept(FunctionNode.java:405)
        at jdk.scripting.nashorn/jdk.nashorn.internal.ir.LexicalContextNode$Acceptor.accept(LexicalContextNode.java:57)
        at jdk.scripting.nashorn/jdk.nashorn.internal.ir.LexicalContextExpression.accept(LexicalContextExpression.java:47)
        at jdk.scripting.nashorn/jdk.nashorn.internal.ir.FunctionNode.accept(FunctionNode.java:57)
        at jdk.scripting.nashorn/jdk.nashorn.internal.codegen.CompilationPhase.transformFunction(CompilationPhase.java:635)
        at jdk.scripting.nashorn/jdk.nashorn.internal.codegen.CompilationPhase$LoweringPhase.transform(CompilationPhase.java:78)
        at jdk.scripting.nashorn/jdk.nashorn.internal.codegen.CompilationPhase.apply(CompilationPhase.java:631)
        at jdk.scripting.nashorn/jdk.nashorn.internal.codegen.Compiler.compile(Compiler.java:657)
        at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.Context.compile(Context.java:1546)
        at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.Context.compileScript(Context.java:1467)
        at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.Context.evaluateSource(Context.java:1433)
        at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.Context.load(Context.java:962)
        at jdk.scripting.nashorn/jdk.nashorn.internal.objects.Global.load(Global.java:1720)
        at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$113$1552AAAAA$jvm_npm.L:17#_load(jar:file:/Users/jerry/.m2/repository/io/reactiverse/es4x/0.7.3/es4x-0.7.3.jar!/io/reactiverse/es4x/jvm-npm.js:92)
        at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$169$4366AA$jvm_npm.L:17#Require(jar:file:/Users/jerry/.m2/repository/io/reactiverse/es4x/0.7.3/es4x-0.7.3.jar!/io/reactiverse/es4x/jvm-npm.js:143)
        at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$137$1464A$jvm_npm.L:17#Module#require(jar:file:/Users/jerry/.m2/repository/io/reactiverse/es4x/0.7.3/es4x-0.7.3.jar!/io/reactiverse/es4x/jvm-npm.js:46)
        at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$404$1AAAAA$fetch/0x00000008004d0040.L:1(node_modules/apollo-server-env/dist/polyfills/fetch.js:3)
        at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptFunctionData.invoke(ScriptFunctionData.java:665)
        at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptFunction.invoke(ScriptFunction.java:513)
        at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptRuntime.apply(ScriptRuntime.java:527)
        at jdk.scripting.nashorn/jdk.nashorn.internal.objects.NativeFunction.apply(NativeFunction.java:104)
        at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$113$1552AAAAA$jvm_npm.L:17#_load(jar:file:/Users/jerry/.m2/repository/io/reactiverse/es4x/0.7.3/es4x-0.7.3.jar!/io/reactiverse/es4x/jvm-npm.js:97)
        at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$169$4366AA$jvm_npm.L:17#Require(jar:file:/Users/jerry/.m2/repository/io/reactiverse/es4x/0.7.3/es4x-0.7.3.jar!/io/reactiverse/es4x/jvm-npm.js:143)
        at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$137$1464A$jvm_npm.L:17#Module#require(jar:file:/Users/jerry/.m2/repository/io/reactiverse/es4x/0.7.3/es4x-0.7.3.jar!/io/reactiverse/es4x/jvm-npm.js:46)
        at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$132$1AAAAA$index$cu1$restOf/0x0000000800414c40.L:1(node_modules/apollo-server-env/dist/index.js:9)
        at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$113$1552AAAAA$jvm_npm.L:17#_load(jar:file:/Users/jerry/.m2/repository/io/reactiverse/es4x/0.7.3/es4x-0.7.3.jar!/io/reactiverse/es4x/jvm-npm.js:97)
        at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$106$4366AA$jvm_npm.L:17#Require(jar:file:/Users/jerry/.m2/repository/io/reactiverse/es4x/0.7.3/es4x-0.7.3.jar!/io/reactiverse/es4x/jvm-npm.js:143)
        at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$88$1464A$jvm_npm.L:17#Module#require(jar:file:/Users/jerry/.m2/repository/io/reactiverse/es4x/0.7.3/es4x-0.7.3.jar!/io/reactiverse/es4x/jvm-npm.js:46)
        at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$128$1AAAAA$index$cu1$restOf.L:1(node_modules/apollo-server-core/dist/index.js:9)
        at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$113$1552AAAAA$jvm_npm.L:17#_load(jar:file:/Users/jerry/.m2/repository/io/reactiverse/es4x/0.7.3/es4x-0.7.3.jar!/io/reactiverse/es4x/jvm-npm.js:97)
        at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$106$4366AA$jvm_npm.L:17#Require(jar:file:/Users/jerry/.m2/repository/io/reactiverse/es4x/0.7.3/es4x-0.7.3.jar!/io/reactiverse/es4x/jvm-npm.js:143)
        at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$88$1464A$jvm_npm.L:17#Module#require(jar:file:/Users/jerry/.m2/repository/io/reactiverse/es4x/0.7.3/es4x-0.7.3.jar!/io/reactiverse/es4x/jvm-npm.js:46)
        at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$115$1AAAAA$index$cu1$restOf.L:1(build/src/apollo-server-es4x/index.js:6)
        at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$113$1552AAAAA$jvm_npm$cu1$restOf.L:17#_load(jar:file:/Users/jerry/.m2/repository/io/reactiverse/es4x/0.7.3/es4x-0.7.3.jar!/io/reactiverse/es4x/jvm-npm.js:97)
        at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$106$4366AA$jvm_npm$cu1$restOf.L:17#Require(jar:file:/Users/jerry/.m2/repository/io/reactiverse/es4x/0.7.3/es4x-0.7.3.jar!/io/reactiverse/es4x/jvm-npm.js:143)
        at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$88$1464A$jvm_npm.L:17#Module#require(jar:file:/Users/jerry/.m2/repository/io/reactiverse/es4x/0.7.3/es4x-0.7.3.jar!/io/reactiverse/es4x/jvm-npm.js:46)
        at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$97$dev.L:1(build/src/dev.js:3)
        at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$95$1552AAZAA$jvm_npm$cu1$restOf.L:17#_load(jar:file:/Users/jerry/.m2/repository/io/reactiverse/es4x/0.7.3/es4x-0.7.3.jar!/io/reactiverse/es4x/jvm-npm.js:97)
        at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$79$3510A$jvm_npm$cu1$restOf.L:17#runMain(jar:file:/Users/jerry/.m2/repository/io/reactiverse/es4x/0.7.3/es4x-0.7.3.jar!/io/reactiverse/es4x/jvm-npm.js:110)
        at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptFunctionData.invoke(ScriptFunctionData.java:657)
        at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptFunction.invoke(ScriptFunction.java:513)
        at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptRuntime.apply(ScriptRuntime.java:527)
        at jdk.scripting.nashorn/jdk.nashorn.api.scripting.ScriptObjectMirror.call(ScriptObjectMirror.java:120)
        at io.reactiverse.es4x.impl.nashorn.NashornRuntime.invokeMethod(NashornRuntime.java:116)
        at io.reactiverse.es4x.impl.nashorn.NashornRuntime.main(NashornRuntime.java:89)
        at io.reactiverse.es4x.JSVerticleFactory$1.start(JSVerticleFactory.java:79)
        at io.vertx.core.impl.DeploymentManager.lambda$doDeploy$8(DeploymentManager.java:552)
        at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:320)
        at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:404)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:462)
        at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:897)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:834)
```

Alias directive:
``` 
"es4xAlias": {
    "node-fetch/lib/index.js": "./build/src/node-fetch-es5/index.js"
  }
```

**Comment by mst209** [comment]
I am having this issue too!  

@pmlopes Any clue would be awesome on this one.

**Comment by pmlopes** [comment]
hi, can you share a small reproducer? in the meantime have you seen? 

https://vertx-graphql.github.io/ 

also note that for the upcoming 0.8.0 release the plan is to drop ashore and go with graaljs only. graaljs works on jdk8 too and on jdk11 it offers the same performance as graaljs itself.

there you've full es2019 support...

**Comment by pmlopes** [comment]
@jerrywgray @mst209 can you provide a simple example showing this issue so I can try to debug?

**Comment by jerrywgray** [comment]
So sorry for the delay. I had to move on to another project for a while.

I updated es4x and it doesn't have any problem with ES6 stuff now which is convenient. However due to the lack of internal node module support I'll still have to alias some things(like it can't load "stream" which is required by node-fetch).

Created an example repo that didn't work. If the file is the same name and local to it then by npm import rules it should grab that. Which did seem to work. When specialStream.js was named stream.js and adjacent to the index.js it loaded it instead of the internal stream.

However any renaming of the file and trying to use the aliasing it doesn't seem to be working. Including loading from a different folder.
https://github.com/jerrywgray/test-es4x-alias

And regarding the graphql package, if it had stitching support I could use it... I can't find the github repo of it to start looking at it to modify. We have another external service that provides a graphql endpoint but we need to extend that endpoint where I can add pub/sub to the vertx event bus. I can't find any java native graphql packages that handle stitching either so I was hoping I could just run our current apollo/express stitching API under vertx. I imagine with enough aliasing I can get it up and running. With Apollo currently pushing their new federation specification it looks like there's a java package getting implemented so depending on time frames this may end up moot. 

It would be nice to be able to have new devs jump in and write verticles in javascript so I'm still looking at using es4x for other worker verticles if I can figure out what the limitations are with third party npm packages.
